File: app/src/main/java/com/icerrate/popularmovies/view/movies/detail/MovieDetailFragmentListener.java
package com.icerrate.popularmovies.view.movies.detail;

import android.view.View;

/**
 * @author Ivan Cerrate.
 */

public interface MovieDetailFragmentListener {

    void setCollapsingTitle(String title);

    void setBackdropImage(String backdropUrl);

    void setFavoriteOnClickListener(View.OnClickListener onClickListener);

    void setFavoriteState(boolean isFavorite);

    void updateFavoriteState(boolean isFavorite);
}

File: app/src/main/java/com/icerrate/popularmovies/view/movies/detail/MovieDetailFragmentListener.kt
package com.icerrate.popularmovies.view.movies.detail

import android.view.View

/**
 * @author Ivan Cerrate.
 */
interface MovieDetailFragmentListener {

    fun setCollapsingTitle(title: String)

    fun setBackdropImage(backdropUrl: String)

    fun setFavoriteOnClickListener(onClickListener: View.OnClickListener)

    fun setFavoriteState(isFavorite: Boolean)

    fun updateFavoriteState(isFavorite: Boolean)
}